initial commit for nuxt server (API mission/list)

Morbakos opened the pull request.

Morbakos committed:
initial commit for nuxt server (API mission/list)

oxypomme committed:
fix(nuxt): :lock: fixed prvate env vars gooing public

oxypomme committed:
docs(nuxt): :memo: added new env vars into sample-env

oxypomme committed:
fix(nuxt): :bug: fixed mission table

oxypomme committed:
docs(nuxt): :wastebasket: marked as deprecated old API urls

Morbakos committed:
add back for show mission

oxypomme committed:
feat(nuxt): :lipstick: added style for mission show

oxypomme committed:
fix(nuxt): :lipstick: fixed error page

oxypomme committed:
fix(nuxt): :lipstick: fixed center issue on mission show

oxypomme committed:
fix(nuxt): :bug: fixed mission image when not loading

oxypomme committed:
build(deps): :arrow_up: upgraded dependencies

Morbakos committed:
add check for mission upload

oxypomme commented on nuxt-client/nuxt.config.ts:
```diff
@@ -13,10 +13,26 @@ export default defineNuxtConfig({
+    MONGO_COLLECTION: process.env.MONGO_COLLECTION_NAME,
+    AUTH0_AUDIENCE: process.env.AUTH0_AUDIENCE,
+    PBO_MANAGER: process.env.PBO_MANAGER,
+    UPLOAD_TEMP_DIR: process.env.TEMP_DIR!.replace(/(?:\r\n|\r|\n)/g, ''),
```
Why not using `path.resolve` ?

oxypomme committed:
fix(nuxt): :label: fixed typing issues

oxypomme committed:
fix(nuxt): :fire: removed useless doublee call to upload

oxypomme committed:
feat(nuxt): :zap: improvements on mission upload

fixed failed error detection in front, fixed filename in back, async/await back, dummy report/mission if PBOManager not defined

oxypomme committed:
fix(nuxt): :bug: fixed repo link

oxypomme committed:
fix(nuxt): :bug: no longer redirect when getAccessToken fails

oxypomme committed:
fix(nuxt): :bug: fixed minor issues with asyncFileExist

Morbakos committed:
Commit de parano

Commit:
switch to localStorage for auth0 token & reactivate auth on add page

Commit:
almost done for upload

Commit:
Make Oxy happy for once :P

Commit:
commit de parano

Commit:
add method to copy collection

Commit:
Fix wrong variable in config

Commit:
finalize mission upload

Commit:
Adapt mission list to old & new model for briefing

oxypomme committed:
feat(nuxt): :sparkles: db client is now a singleton

oxypomme committed:
feat(nuxt): :sparkles: added compact list

oxypomme committed:
perf(nuxt): :zap: better indexing in table

oxypomme committed:
feat(nuxt): :sparkles: added filters for mission list

oxypomme committed:
fix(nuxt): :bug: added minimum for players filters

oxypomme committed:
feat(nuxt): :sparkles: added fuzzy search in list

oxypomme committed:
refactor(nuxt): :recycle: refactored usage of useRuntimeConfig & AUTH0_AUDIENCE

oxypomme committed:
build(nuxt): :wrench: now using nuxt overloading env

oxypomme committed:
fix(nuxt): :bug: fixed boolean & number filters at 0/false

oxypomme committed:
fix(nuxt): :bug: minor fixs in mission list

oxypomme committed:
feat(nuxt): :sparkles: using v-tooltip instead of titles

oxypomme committed:
refactor(nuxt): :fire: removed unused component

oxypomme committed:
feat(nuxt): :sparkles: added clear on drop down filters

Morbakos committed:
Remove unused route

Morbakos committed:
Add enpoint to update mission

Morbakos committed:
(Non-working) commit for exam

oxypomme committed:
fix(nuxt): :bug: fixed set playable route

also refactored to be more coherent, and upgraded deps

oxypomme committed:
fix(nuxt): :passport_control: somehow fixed auth0

oxypomme committed:
fix(nuxt): :bug: fixed mission not beign set a playable

oxypomme committed:
chore: :truck: deleted old files, moved new app, added docker

oxypomme committed:
docs: :memo: fixed missing commands for dev

oxypomme committed:
chore: :technologist: updated scopes
